hazelcast/src/main/java/com/hazelcast/sql/impl/expression/predicate/ComparisonPredicate.java

```diff
@@ -77,6 +78,13 @@ public Boolean eval(Row row, ExpressionEvalContext context) {
             return null;
         }
 
+        Class<?> leftClass = left.getClass();
+        Class<?> rightClass = right.getClass();
+
+        if (!(left instanceof Comparable && leftClass.equals(rightClass))) {
```

Comment:
The check for class should be performed only for operands of `OBJECT` type since for all other operand types, we know for sure that they are comparable, and of the same type.